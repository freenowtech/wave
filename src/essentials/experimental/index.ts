export * from './theme';
export { createGlobalStyle, ColorPalette } from './Colors';
export { darkTheme } from './globalStyles';
export { getSemanticValue, getBareValue, getBareHslValue, getSemanticHslValue } from './cssVariables';
