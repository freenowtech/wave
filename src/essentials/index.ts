export { Elevation } from './Elevation/Elevation';
export { Spaces } from './Spaces/Spaces';
export { Breakpoints, MediaQueries } from './Breakpoints/Breakpoints';
export type { ReadCssColorVariable, ReadCssVariable } from './Colors/types';
export { GlobalStyle as ClassicColors } from './Colors/Colors';
export { GlobalStyle as ModernColors } from './Colors/ModernColors';
