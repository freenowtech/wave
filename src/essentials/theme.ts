import { Colors, SemanticColors } from './Colors/Colors';
import { Spaces } from './Spaces/Spaces';
import { Breakpoints, MediaQueries } from './Breakpoints/Breakpoints';

const theme = {
    breakpoints: Breakpoints,
    colors: Colors,
    // todo: rename semanticColors to colors in the next major release
    semanticColors: SemanticColors,
    fontSizes: ['0.75rem', '0.875rem', '1rem', '1.25rem', '1.5rem', '2rem', '2.5rem', '3rem'],
    fontWeights: {
        normal: 400,
        semibold: 600,
        bold: 700
    },
    fonts: {
        normal: fontStack(['Open Sans', 'sans-serif']),
        monospace: fontStack(['Hack', 'monospace'])
    },
    mediaQueries: MediaQueries,
    space: Spaces,
    radii: ['0rem', '0.125rem', '0.25rem', '0.5rem'],
    iconSizes: {
        small: 12,
        medium: 24,
        large: 28
    },
    shadows: {
        small: '0 0.0625rem 0.25rem 0 rgba(0,0,0,0.1)',
        medium: '0 0 0.5rem 0.1875rem rgba(0,0,0,0.08)',
        large: '0 0 0.75rem 0.25rem rgba(0,0,0,0.12)'
    }
};

/**
 * Custom aliases declaration
 * https://styled-system.com/theme-specification/#scale-aliases
 */

// @ts-ignore
theme.fontSizes.small = theme.fontSizes[1];
// @ts-ignore
theme.fontSizes.medium = theme.fontSizes[2];
// TODO: the alias large should match with scale on theme "fontSizes" but it may require a breaking change
// @ts-ignore
theme.fontSizes.large = '1.125rem';

function fontStack(fonts) {
    return fonts.map(font => (font.includes(' ') ? `"${font}"` : font)).join(', ');
}

export { theme };
