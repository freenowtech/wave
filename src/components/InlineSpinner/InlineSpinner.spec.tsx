import { render } from '@testing-library/react';
import React from 'react';
import { InlineSpinner } from '..';
import { Colors } from '../../essentials';

describe('InlineSpinner', () => {
    it('renders the default variant', () => {
        const { container } = render(<InlineSpinner />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the small size', () => {
        const { container } = render(<InlineSpinner size="small" />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders in a custom color', () => {
        const { container } = render(<InlineSpinner color={Colors.POSITIVE_GREEN_900} />);

        expect(container.firstChild).toMatchSnapshot();
    });
});
