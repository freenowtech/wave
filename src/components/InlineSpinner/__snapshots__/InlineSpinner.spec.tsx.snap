// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineSpinner renders in a custom color 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  width: 1.25rem;
  height: 1.25rem;
  vertical-align: text-bottom;
  border: 0.125rem solid;
  border-right-color: transparent;
  border-radius: 50%;
  -webkit-animation: wdnds 750ms linear infinite;
  animation: wdnds 750ms linear infinite;
}

<span>
  <div
    class="c0"
  />
</span>
`;

exports[`InlineSpinner renders the default variant 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  width: 1.25rem;
  height: 1.25rem;
  vertical-align: text-bottom;
  border: 0.125rem solid;
  border-right-color: transparent;
  border-radius: 50%;
  -webkit-animation: wdnds 750ms linear infinite;
  animation: wdnds 750ms linear infinite;
}

<span>
  <div
    class="c0"
  />
</span>
`;

exports[`InlineSpinner renders the small size 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  width: 1.25rem;
  height: 1.25rem;
  vertical-align: text-bottom;
  border: 0.125rem solid;
  border-right-color: transparent;
  border-radius: 50%;
  -webkit-animation: wdnds 750ms linear infinite;
  animation: wdnds 750ms linear infinite;
}

<span>
  <div
    class="c0"
  />
</span>
`;
